File: 2089.find-target-indices-after-sorting-array.java
import java.util.*;

class Solution {
    public List<Integer> targetIndices(int[] nums, int target) 
    {
        boolean isPresent = false;
        int maxElement = Arrays.stream(nums).max().getAsInt();
        int[] newNums = new int[maxElement+1];

        for(int val: nums)
        {
            newNums[val]++;
            if(val == target)
            isPresent = true;
        }
        
        
        for( int i = 1 ; i < newNums.length ;i++)
        {
            newNums[i] = newNums[i] + newNums[i-1];
        }

        if(!isPresent)
        {
            return new ArrayList<Integer>();
        }
        else
        {
            int[] theFreq = new int[newNums[target]-newNums[target-1]];
        
        

        int j = theFreq.length -1;
        while((newNums[target]-newNums[target-1]) > 0)
        {
            theFreq[j] = --newNums[target];
            j--;
        }

        List<Integer> theList = new ArrayList<Integer>();

        for ( int val: theFreq)
        theList.add(val);
        
        return theList;

        }
        
    }
}

File: 2089-find-target-indices-after-sorting-array/2089-find-target-indices-after-sorting-array.cpp
class Solution {
public:
    vector<int> targetIndices(vector<int>& nums, int target) 
    {
        vector<int> freq(101);   
        for(int i = 0; i < nums.size();i++){freq[nums[i]] += 1;}
        for(int i = 1; i < freq.size();i++){freq[i] = freq[i] + freq[i-1];}
        vector<int> solution;
        if(freq[target] > 0)
        {
            for(int i = freq[target-1] ; i < freq[target];i++)                                                  {solution.insert(solution.end(),i);}
        }
        else{return solution;}
        
        return solution;
    }
};